src/NServiceBus.Transport.RabbitMQ/Connection/ChannelProvider.cs

```diff
@@ -21,36 +21,45 @@ public ChannelProvider(ConnectionFactory connectionFactory, TimeSpan retryDelay,
 
         public void CreateConnection()
         {
+            if (connection is not null)
+            {
+                connection.ConnectionShutdown -= Connection_ConnectionShutdown;
+                connection.Dispose();
+                connection = null;
```

Comment:
There is really no reason to do this, considering the very next line in is to reassign this anyway.

Reply:
Without this we could reliably reproduce races against the field leading to mask the original intended exception no longer being raised but Rsmon has more context

Reply:
We were not sure if the event unsubscribing was needed but the Dispose + null is definitely required to prevent shutdown race conditions as the call on the next like can take several seconds when the remote is unavailable and fail.